backend/internal/database/book.go

```diff
@@ -357,3 +357,123 @@ func (db *BookDatabase) FetchBookAnalytics(ctx context.Context, id string, dRang
 	return analytics, err
 
 }
+
+/*
+ * Returns an array containing the favorite books for the given userID
+ */
+func (db *BookDatabase) FetchFavorites(ctx context.Context, userID string) ([]models.Book, error) {
+
+	var favoriteBooks []models.Book
+	favoriteBooks = make([]models.Book, 0)
+
+	var query string = "SELECT book_id FROM favorite_books WHERE user_id = $1"
+
+	rows, err := db.Conn.Query(ctx, query, userID)
+
+	if err != nil {
+		fmt.Print(err)
+		return nil, errors.Wrap(err, "error on SELECT FROM favorite_books in FetchFavorites")
+	}
+
+	defer rows.Close()
+
+	// Iterate through all rows of query (which contain userID)
+	for rows.Next() {
+
+		var bookID string
+
+		// Extract bookIDs from rows
+		if err := rows.Scan(&bookID); err != nil {
+			fmt.Print(err)
+			return nil, errors.Wrap(err, "error scanning result of"+
+				" SELECT FROM favorite_books in FetchFavorites")
+		}
+
+		// Query books table to obtain pointer to book object
+		book, err2 := db.FetchBook(ctx, bookID)
+
+		if err2 != nil {
+			fmt.Print(err2)
+			return nil, errors.Wrap(err, "error on FetchBook from FetchFavorites")
+		}
+
+		// Add favorite book to array
+		favoriteBooks = append(favoriteBooks, *book)
+
+	}
```

Comment:
This is needlessly [N+1](https://secure.phabricator.com/book/phabcontrib/article/n_plus_one/)

Can you combine this into the `SELECT FROM favorite_books` query by way of a join or `book.id IN` test?